spec/api-power-monitor-spec.js

```diff
@@ -85,5 +85,34 @@ const skip = process.platform !== 'linux' || !process.env.DBUS_SYSTEM_BUS_ADDRES
         })
       })
     })
+
+    describe('when a listener is added to shutdown event', () => {
+      before(async () => {
+        const calls = await getCalls()
+        assert.equal(calls.length, 2)
+        powerMonitor.once('shutdown', () => { })
+      })
+
+      it('should call Inhibit to delay shutdown', async () => {
+        const calls = await getCalls()
+        assert.equal(calls.length, 3)
+        assert.deepEqual(calls[2].slice(1), [
+          'Inhibit', [
+            [[{type: 's', child: []}], ['shutdown']],
+            [[{type: 's', child: []}], ['electron']],
+            [[{type: 's', child: []}], ['Ensure a clean shutdown']],
+            [[{type: 's', child: []}], ['delay']]
+          ]
+        ])
+      })
+
+      describe('when PrepareForShutdown(true) signal is sent by logind', () => {
+        it('should emit "shutdown" event', (done) => {
+          powerMonitor.once('shutdown', () => { done() })
+          emitSignal('org.freedesktop.login1.Manager', 'PrepareForShutdown',
+            'b', [['b', true]])
+        })
+      })
+    })
```

Comment:
> What are the two calls that happen prior to adding the listener?

As I explained to @alexeykuzmin:

>  I also added an assertion in the before callback to make it explicit that getCalls returns two calls prior to adding the listener.

> If we keep the preventDefault approach, please add a test here to exercise it

I don't see how I could test this since there's no way of checking if a file descriptor was closed. (python-dbusmock doesn't even send a real file descriptor when it returns from an `Inhibit` request)